src/main/java/org/terasology/taskSystem/actions/PerformTaskNode.java

```diff
@@ -122,4 +205,354 @@ private void changeOreonAttributes(Actor oreon, TaskComponent taskComponent) {
 
         oreon.save(oreonAttributeComponent);
     }
+
+    /**
+     * Places the required blocks in the selected area based on the task selected
+     * @param oreon The Oreon entity working on the task
+     * @param taskComponent The component with task information which just completed
+     */
+    private void completeTask(TaskComponent taskComponent) {
+        Region3i selectedRegion = taskComponent.taskRegion;
+        String taskType = taskComponent.assignedTaskType;
+
+
+        switch (taskType) {
+            case AssignedTaskType.Plant :
+                placeCrops(selectedRegion, Constants.OREON_CROP_PREFAB);
+                break;
+
+            case AssignedTaskType.Build :
+                //constructBuilding(selectedRegion, taskComponent.buildingType);
+                constructBuildingUsingStructureTemplates(selectedRegion, taskComponent.buildingType);
+        }
+    }
+
+    private void placeCrops(Region3i selectedRegion, String cropToPlace) {
+        int minX = selectedRegion.minX();
+        int maxX = selectedRegion.maxX();
+        int minZ = selectedRegion.minZ();
+        int maxZ = selectedRegion.maxZ();
+
+        int y = selectedRegion.minY();
+
+        for (int x = minX; x <= maxX; x++) {
+            for (int z = minZ; z <= maxZ; z++) {
+                Block block = blockManager.getBlock(cropToPlace);
+                blockEntityRegistry.setBlockForceUpdateEntity(new Vector3i(x, y + 1, z), block);
+            }
+        }
+    }
+
+    private void constructBuilding(Region3i selectedRegion, BuildingType buildingType) {
```

Comment:
Move class to a separate class for constructing buildings task.
Move building-specific code to a separate class.  Possibly the raster class.